File: lib/nic/exec/xsd/validator.py
#!/usr/bin/env python

from lxml import etree


def validate(schema_file, data_file):

    data    = etree.parse(open(data_file))
    schema  = etree.XMLSchema(etree.parse(open(schema_file)))

    schema.assertValid(data)


if __name__ == '__main__':

    import sys

    try:
        schema_file, data_file = sys.argv[1:3]
    except:
        print ("Usage: %s <schema-file> <data-file>" % sys.argv[0])
        sys.exit(128)

    try:
        validate(schema_file, data_file)
        print("%s: %s is Valid." % (schema_file, data_file))
        sys.exit(0)

    except etree.DocumentInvalid:
        print("%s: %s is NOT VALID!" % (schema_file, data_file))
        sys.exit(1)

File: lib/api/nic/exec/xsd/validator.php
#!/usr/bin/env php
<?php
# vim: set autoindent tabstop=8 softtabstop=4 expandtab shiftwidth=4

if (count($argv) != 3)
{
    print "Usage: {$argv[0]} <schema-file> <data-file>\n";
    exit(128);
}

$schema_file = $argv[1];
$data_file = $argv[2];

$dom = new DOMDocument();
$dom->load($data_file);
$v = $dom->schemaValidate($schema_file);

if ($v) {
    print "$schema_file: $data_file is Valid.\n";
    exit(0);
}
else {
    print "$schema_file: $data_file is NOT VALID!\n";
    exit(1);
}

